ext/EnzymeSparseArraysExt.jl

```diff
@@ -0,0 +1,123 @@
+module EnzymeSparseArraysExt
+
+using Enzyme
+using EnzymeCore
+using EnzymeCore: EnzymeRules
+
+using SparseArrays
+using SparseArrays: LinearAlgebra, SparseMatrixCSCUnion
+const SparseMatAdj = Union{SparseMatrixCSC, LinearAlgebra.Adjoint{T, SparseMatrixCSC} where T}
+
+# TODO don't limit A to be Const. Currently I'd have to implement a new matmul for this
+# or bootstrap ChainRules ProjectTo mechanism to enforce the structural zeros
+# Currently we put the rule on the 5-arg mul!, since spdensemul! isn't a public API and I am unsure
+# how stable it is.
+function EnzymeRules.augmented_primal(config::EnzymeRules.ConfigWidth, 
+                                      func::Const{typeof(LinearAlgebra.mul!)},
+                                      ::Type{RT}, 
+                                      C::Annotation{<:StridedVecOrMat},
+                                      A::Const{<:SparseMatAdj},
+                                      B::Annotation{<:StridedVecOrMat},
+                                      α::Annotation{<:Number},
+                                      β::Annotation{<:Number}
+                                    ) where {RT}
+
+    # Q? Why doesn't EnzymeRules.overwritten(config) detect that C is overwritten?
```

Comment:
overwritten doesn't consider the impact of the current function (just code which occurs after the call)

Reply:
That makes a ton of sense.